bibtex-completion.el

```diff
@@ -478,6 +493,23 @@ for string replacement."
 (defvar bibtex-completion-cached-notes-keys nil
   "A cache storing notes keys obtained when the bibliography was last parsed.")
 
+(defun bibtex-completion--read ()
+  "Select BibTeX entries in completion system."
+  ;; define a completion function that defaults to completing-read, but can be overridden
+  (list (cdr (assoc (completing-read-bibtex--read) (bibtex-completion--get-candidates)))))
+
+(defun bibtex-completion--completing-read ()
+  "Read bibtex-completion entries for completion using completing-read."
+  (bibtex-completion-init)
+  (completing-read
+   "BibTeX entries: "
+   (bibtex-completion--get-candidates)))
+
+(defun bibtex-completion--get-candidates ()
+  "Return all keys from bibtex-completion-candidates."
+  (mapcar (lambda (cand) (cons
+                          (bibtex-completion-format-entry cand (1- (frame-width))) (cdr (assoc "=key="                                                                                                    cand)))) (bibtex-completion-candidates)))
```

Comment:
We need a better solution for this, as we're missing data in the process.

If we can't complete against the output of `bibtex-completion-candidates` using `completing-read`, can we transform it so we don't lose any of that data?

The use case here is I really want to add tags to complete against, so I can distinguish among different projects, etc. That works in `ivy-bibtex` and `helm-bibtex`, but not here.

See #362.

Reply:
I think the answer is on the selectrum wiki:

https://github.com/raxod502/selectrum/wiki/Tips-for-Creating-Commands